jssource/jsmin.php

```diff
@@ -45,4 +45,50 @@ protected function jsParser() {
         require_once('jssource/Minifier.php');
         return Minifier::minify($this->text);
 	}
+
+    /**
+     * Join and minify JS files
+     *
+     * @param array $jsFiles an 'array' of js files
+     *
+     * @author Jose C. Massón <jose@gcoop.coop>
+     *
+     * @return Minified JS file path
+     */
+    static public function joinAndMinifyJSFiles($jsFiles)
+    {
+        $target = SugarThemeRegistry::current()->getJSPath()
+                . '/' .
+                sha1(implode('|', $jsFiles)) . '.js';
```

Comment:
Hi @Abuelodelanada,

yet again, thanks for this contribution, very cool.

When reviewing the code, two question came to mind:

- If two different modules have a file with the same name, like `edit.js` , won't they both hit the same cached file? thus overwriting one another?

- Another thing, about the ThemeCache, I'm not sure if all files are removed from the theme js cache, when doing a Repair and Rebuild. I just gave a quick look at the repair and rebuild code, so maybe wrong.
    - When testing, did the changes to existing js files get onto the cache and get refreshed on the browser?

Reply:
Hi @clemente-raposo 

Thanks for the feedback!

> * If two different modules have a file with the same name, like edit.js , won't they both hit the same cached file? thus overwriting one another?

You are right. I'll fix it!

> * Another thing, about the ThemeCache, I'm not sure if all files are removed from the theme js cache, when doing a Repair and Rebuild. I just gave a quick look at the repair and rebuild code, so maybe wrong.

I'll check it

> * When testing, did the changes to existing js files get onto the cache and get refreshed on the browser?

When testing you should delete cache files

Reply:
Done in commit 125dadf36c6b4aabf9ccf44e019405f60651c7f2

Reply:
@gymad when you talk about the "js grouping functionality" you mean the file `jssource/JSGroupings.php` ??